src/data/types.rs

```diff
@@ -281,113 +281,165 @@ pub(super) fn print_type(
+                ));
+
+                if needs_parens {
+                    postfixes.push(")".to_string());
+                } else {
+                    postfixes.push(String::new());
+                }
             }
-            Ok(())
+            _ => unreachable!(),
         }
     }
-}
-fn print_post(ctype: &Type, f: &mut Formatter) -> fmt::Result {
-    match ctype {
-        Type::Pointer(to, _) => print_post(to, f),
-        Type::Array(to, size) => {
-            write!(f, "[")?;
-            if let ArrayType::Fixed(size) = size {
-                write!(f, "{}", size)?;
-            }
-            write!(f, "]")?;
-            print_post(to, f)
+
+    let final_type = unrolled_type[unrolled_type.len() - 1];
+    match final_type {
+        Char(signed) | Short(signed) | Int(signed) | Long(signed) => {
+            write!(
+                f,
+                "{}{}",
+                if *signed { "" } else { "unsigned " },
+                match final_type {
+                    Char(_) => "char",
+                    Short(_) => "short",
+                    Int(_) => "int",
+                    Long(_) => "long",
+                    _ => unreachable!(),
+                }
+            )?;
         }
-        Type::Function(func_type) => {
-            write!(f, "(")?;
-            let mut params = func_type.params.iter();
-            let print = |f: &mut _, symbol: MetadataRef| {
-                let symbol = symbol.get();
-                let id = if symbol.id == InternedStr::default() {
-                    None
-                } else {
-                    Some(symbol.id)
-                };
-                print_type(&symbol.ctype, id, f)
-            };
-            if let Some(&first) = params.next() {
-                print(f, first)?;
-            }
-            for &symbol in params {
-                write!(f, ", ")?;
-                print(f, symbol)?;
-            }
-            if func_type.varargs {
-                write!(f, ", ...")?;
-            }
-            write!(f, ")")
+        Bool => write!(f, "_Bool")?,
+        Float => write!(f, "float")?,
+        Double => write!(f, "double")?,
+        Void => write!(f, "void")?,
+        Enum(Some(ident), _) => write!(f, "enum {}", ident)?,
+        Enum(None, _) => write!(f, "<anonymous enum>")?,
+        Union(struct_type) => {
+            write!(f, "union ")?;
+            write_struct_type(struct_type, f)?;
+        }
+        // Union(StructType::Named(ident, _)) => format!("union {}", ident),
+        // Union(_) => "<anonymous union>".to_string(),
+        Struct(struct_type) => {
+            write!(f, "struct ")?;
+            write_struct_type(struct_type, f);
         }
-        _ => Ok(()),
+        // Struct(StructType::Named(ident, _)) => format!("struct {}", ident),
+        // Struct(_) => "<anonymous struct>".to_string(),
+        VaList => write!(f, "va_list")?,
+        Error => write!(f, "<type error>")?,
+        _ => unreachable!(),
```

Comment:
Can you write out the declarators explicitly so it's more clear why they're unreachable? And maybe a comment as well.
```suggestion
        Array(_, _) | Function(_) | Pointer(_, _) => unreachable!("declarators were handled above"),
```